clippy_lints/src/swap.rs

```diff
@@ -172,15 +172,16 @@ fn check_manual_swap(cx: &LateContext<'_>, block: &Block<'_>) {
     }
 }
 
+#[allow(clippy::too_many_lines)]
 /// Implementation of the `ALMOST_SWAPPED` lint.
 fn check_suspicious_swap(cx: &LateContext<'_>, block: &Block<'_>) {
-    for w in block.stmts.windows(2) {
+    for [first, second] in block.stmts.array_windows() {
         if_chain! {
-            if let StmtKind::Semi(first) = w[0].kind;
-            if let StmtKind::Semi(second) = w[1].kind;
+            if let StmtKind::Semi(first_semi) = first.kind;
+            if let StmtKind::Semi(second_semi) = second.kind;
             if first.span.ctxt() == second.span.ctxt();
-            if let ExprKind::Assign(lhs0, rhs0, _) = first.kind;
-            if let ExprKind::Assign(lhs1, rhs1, _) = second.kind;
+            if let ExprKind::Assign(lhs0, rhs0, _) = first_semi.kind;
+            if let ExprKind::Assign(lhs1, rhs1, _) = second_semi.kind;
             if eq_expr_value(cx, lhs0, rhs1);
             if eq_expr_value(cx, lhs1, rhs0);
```

Comment:
`parse`/`is_same`  should be replacing the `StmtKind/ExprKind` checks here, not an additional `if`

Reply:
@Alexendoo Replaced these logic, thanks.